### ISSUE TITLE
Validation part handlers only triggered from api controller 

### ISSUE THREAD

**Issue by giannik** [OrchardCMS/OrchardCore#13722]
I would like to use validation part handlers to check eg for various conditions before saving content items  from the admin ui like the following code : 

```
public class LayoutProfilePartHandler : ContentPartHandler<LayoutProfilePart>
{
    private readonly ISession _session;
    private readonly IStringLocalizer S;

    public LayoutProfilePartHandler(
        ISession session,
        IStringLocalizer<LayoutProfilePartHandler> stringLocalizer)
    {
        _session = session;
        S = stringLocalizer;
    }

    public override async Task ValidatingAsync(ValidateContentContext context, LayoutProfilePart part)
    {

        var boundContentType = part.BoundContentType;
        var boundContentTypeExists = await _session
            .QueryIndex<LayoutProfilePartIndex>(index => index.BoundContentType == boundContentType  &&  index.ContentItemId != part.ContentItem.ContentItemId)
            .CountAsync() > 0;

        if (boundContentTypeExists)
        {
            context.Fail(new ValidationResult(S["The content type '{0}' already exists.", boundContentType], new[] { boundContentType }));
        }
    }
}

```

But this does not work from the admin UI. It seems that this only is triggered from the  contents api controller during post . 
This seems rather cumbersome. 
Why dont the part validation handlers trigger from the admin ui as well. It seems rather inconsistent. 
Also looking at the TitlePart validation handlers there seems to be duplication of validation logic ("A value is required for Title" ) in the TitlePartDriver    and the TitlePart Handler. 
I propose to centralize the validation handlers from ContentPartHandlerCoordinator  to support both validation from the admin ui and api calls.
@MikeAlhayek

**Comment by MikeAlhayek** [comment]
I had also mentioned the same thing during a meeting with @hishamco . I think the main issue in the past we had no way to set the property key for each error but now we do.  I think the `ContentItemDisplayCoordinator` should call the `ValidatingAsync` prior invoking `UpdateAsync` and `ValidatedAsync` is there are no errors.

However, this isn't an easy change and will cause a huge breaking change for everyone since the validation will have to be extracted into a handler for every driver.

Also, this may introduce another challenge, in some controller we may have a driver that populates multiple viewmodels for example

```
var vm1 = new OneViewModel();
var vm2 = new TwoViewModel();

if(await context.Updater.TryUpdateModelAsync(vm1, Prefix) && await context.Updater.TryUpdateModelAsync(vm2, Prefix)) 
{
  // do something.
}
```

Maybe the `Updated` can have `TryValidatePartAsync` that internally calls the `ValidatingAsync` method of the given part. 

@sebastienros / @jtkech is there other reasons why we need to validate in the display driver instead of placing all validation logic in a handler?

**Comment by hishamco** [comment]
Ya I remembered when we talked about that, let us hear @sebastienros feedback, maybe there's a reason for that

**Comment by jtkech** [comment]
Both drivers and handlers were already used but validating before updating was first only done in drivers which are UI oriented (+ mvc binding), we then added handler validations when updating by code (no UI editor) for example for importing and API. Yes not so easy to harmonize, at least we could try to use as much as possible the same validation helpers.

**Comment by MikeAlhayek** [comment]
Yes. The main difference between the two is how we bind data to contentItem.

A good way to do this is to use UpdateAsync on the ContentDisplayManager in the API controller instead of binding directly to ContentItem. This way the drivers logic will always be enforced and the prefixed used in the API will have to be the same as the ones used in the UI. Doing this will be a breaking change since the POST request to the API will have to be changed to match the UI.
